demo_usage/convert_tracking_dataset_to_forecasting.py

```diff
@@ -0,0 +1,196 @@
+	"""
+	if len(instances) == 0:
+		return instances
+	assert isinstance(instances, np.ndarray)
+
+	# create affinity matrix as inverse distance to other objects
+	affinity_matrix = compute_affinity_matrix(copy.deepcopy(instances), copy.deepcopy(instances))
+
+	row_idxs, col_idxs = np.where(affinity_matrix == 0)
+	# find the indices where each row index appears for the first time
+	unique_row_idxs, unique_element_idxs = np.unique(row_idxs, return_index=True)
+	# choose the first instance in each column where repeat occurs
+	first_col_idxs = col_idxs[unique_element_idxs]
+	# eliminate redundant column indices
+	unique_ids = np.unique(first_col_idxs)
+	return instances[unique_ids]
+
+
+def compute_affinity_matrix(dts1: List[Box3d], dts2: List[Box3d]) -> np.ndarray:
+	"""Calculate the affinity matrix between detections and ground truth labels,
+	using a specified affinity function type.
+	
+	Args:
+		dts1: set of detections (N,).
+		dts2: other set of detections (M,).
+	
+	Returns:
+		sims: Affinity scores between detections and ground truth annotations (N, M).
+	"""
+	dt_centers = np.array([dt.translation for dt in dts])
+	gt_centers = np.array([gt.translation for gt in gts])
+	sims = -cdist(dt_centers, gt_centers)
+	return sims
+
+
+def construct_argoverse_boxes_lidarfr(
+	timestamp: int,
+	sweep_labels: List[Dict[str,Any]],
+	city_SE3_egovehicle: SE3
+) -> List[Box3d]:
+	""" Move egovehicle frame boxes, to live in the LiDAR frame instead"""
+	# Make list of Box objects including coord system transforms.
+	box_list = []
+	for label in sweep_labels:
+
+		x = label['center']['x']
+		y = label['center']['y']
+		z = label['center']['z']
+
+		qw = label['rotation']['w']
+		qx = label['rotation']['x']
+		qy = label['rotation']['y']
+		qz = label['rotation']['z']
+
+		obj_label_record = json_label_dict_to_obj_record(label)
+		vertices_egofr = obj_label_record.as_2d_bbox()
+
+		box = Box3d(
+			center = [x, y, z], # Argoverse and nuScenes use scalar-first
+			width = label['width'],
+			length = label['length'],
+			height = label['height'],
+			orientation_q = [qw, qx, qy, qz],
+			name = label['label_class'],
+			track_id = label['track_label_uuid'],
+			timestamp = timestamp
+		)
+
+		# transform box from the egovehicle frame into the city frame
+		box.center = city_SE3_egovehicle.transform_point_cloud(np.array(box.center).reshape(1,3)).squeeze()
+		box.center = list(box.center)
+		box.orientation_q = list(rotmat2quat(city_SE3_egovehicle.rotation @ quat2rotmat(list(box.orientation_q))))
+		vertices_cityfr = city_SE3_egovehicle.transform_point_cloud(vertices_egofr)
+		box.v0 = list(vertices_cityfr[0])
+		box.v1 = list(vertices_cityfr[1])
+		box.v2 = list(vertices_cityfr[2])
+		box.v3 = list(vertices_cityfr[3])
+		box.orientation_yaw = quaternion_yaw_scipy(box.orientation_q)
+		box_list.append(box)
+
+	return box_list
+
+
+def create_tracks_csv(root_path: str, split_subdir: str) -> None:
```

Comment:
[Nit] Generally, we should prefer to use pathlib `Path` objects in place of strings. If the rest of the argoverse-api uses `str`s, then let's just leave this as is.